core/chain_manager_test.go

```diff
@@ -483,22 +485,71 @@ func TestInsertNonceError(t *testing.T) {
 	}
 }
 
-/*
-func TestGenesisMismatch(t *testing.T) {
-	db, _ := ethdb.NewMemDatabase()
-	var mux event.TypeMux
-	genesis := GenesisBlock(0, db)
-	_, err := NewChainManager(genesis, db, db, db, thePow(), &mux)
-	if err != nil {
+func TestChainReorgMissingTransactions(t *testing.T) {
```

Comment:
The confusion probably stems from the fact that I'm using the same key and the reliance of the randomness introduces due to the signing. This is because the ` WriteGenesisBlockForTesting` method only accepts 1 state object.